internal/cli/dot.go

```diff
@@ -0,0 +1,317 @@
+	"github.com/spf13/cobra"
+	"github.com/tmc/dot"
+	"golang.org/x/exp/slices"
+	"golang.org/x/sync/errgroup"
+
+	"chainguard.dev/apko/pkg/build"
+	"chainguard.dev/apko/pkg/build/types"
+)
+
+func dotcmd() *cobra.Command {
+	var extraKeys []string
+	var extraRepos []string
+	var archstrs []string
+	var web, span bool
+
+	cmd := &cobra.Command{
+		Use:   "dot",
+		Short: "Output a digraph showing the resolved dependencies of an apko config.",
+		Long: `Output a digraph showing the resolved dependencies of an apko config.
+
+# Render an svg of example.yaml
+apko dot example.yaml | dot -Tvsg > graph.svg
+
+# Open browser to explore example.yaml
+apko dot --web example.yaml
+
+# Open browser to explore example.yaml, rendering a (almost) minimum spanning tree
+apko dot --web -S example.yaml
+`,
+		Example: `  apko show-packages <config.yaml>`,
+		Args:    cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			archs := types.ParseArchitectures(archstrs)
+			return DotCmd(cmd.Context(), args[0], archs, web, span,
+				build.WithConfig(args[0]),
+				build.WithExtraKeys(extraKeys),
+				build.WithExtraRepos(extraRepos),
+			)
+		},
+	}
+
+	cmd.Flags().StringSliceVarP(&extraKeys, "keyring-append", "k", []string{}, "path to extra keys to include in the keyring")
+	cmd.Flags().StringSliceVarP(&extraRepos, "repository-append", "r", []string{}, "path to extra repositories to include")
+	cmd.Flags().StringSliceVar(&archstrs, "arch", nil, "architectures to build for (e.g., x86_64,ppc64le,arm64) -- default is all, unless specified in config. Can also use 'host' to indicate arch of host this is running on")
+	cmd.Flags().BoolVarP(&span, "spanning-tree", "S", false, "does something like a spanning tree to avoid a huge number of edges")
+	cmd.Flags().BoolVar(&web, "web", false, "launch a browser")
+
+	return cmd
+}
+
+func DotCmd(ctx context.Context, configFile string, archs []types.Architecture, web, span bool, opts ...build.Option) error {
+	wd, err := os.MkdirTemp("", "apko-*")
+	if err != nil {
+		return fmt.Errorf("failed to create working directory: %w", err)
+	}
+	defer os.RemoveAll(wd)
+
+	o, ic, err := build.NewOptions(opts...)
+	if err != nil {
+		return err
+	}
+
+	// cases:
+	// - archs set: use those archs
+	// - archs not set, bc.ImageConfiguration.Archs set: use Config archs
+	// - archs not set, bc.ImageConfiguration.Archs not set: use all archs
+	switch {
+	case len(archs) != 0:
+		ic.Archs = archs
+	case len(ic.Archs) != 0:
+		// do nothing
+	default:
+		ic.Archs = types.AllArchs
+	}
+	// save the final set we will build
+	archs = ic.Archs
+	o.Logger().Infof("Determining packages for %d architectures: %+v", len(ic.Archs), ic.Archs)
+
+	// The build context options is sometimes copied in the next functions. Ensure
+	// we have the directory defined and created by invoking the function early.
+	defer os.RemoveAll(o.TempDir())
+
+	// TODO: Perhaps we want to support multiple architectures?
+	arch := archs[0]
```

Comment:
I don't think we need to, and could probably default to x86_64 and get the same results.

There's no way for an apk to conditionally depend on something based on arch.